src/d3d9/d3d9_common_texture.h

```diff
@@ -366,6 +366,42 @@ namespace dxvk {
     void PreLoadAll();
     void PreLoadSubresource(UINT Subresource);
 
+    void AddUpdateDirtyBox(CONST D3DBOX* pDirtyBox, uint32_t layer) {
+      if (pDirtyBox) {
+        D3DBOX box = *pDirtyBox;
+        if (box.Right <= box.Left
+          || box.Bottom <= box.Top
+          || box.Back <= box.Front)
+          return;
+
+        D3DBOX& updateBox = m_updateDirtyBoxes[layer];
+        if (updateBox.Left == 0 && updateBox.Right == 0
+          && updateBox.Top == 0 && updateBox.Bottom == 0
+          && updateBox.Front == 0 && updateBox.Back == 0) {
+          updateBox = box;
+        } else {
+          updateBox.Left    = std::min(updateBox.Left,   box.Left);
+          updateBox.Right   = std::max(updateBox.Right,  box.Right);
+          updateBox.Top     = std::min(updateBox.Top,    box.Top);
+          updateBox.Bottom  = std::max(updateBox.Bottom, box.Bottom);
+          updateBox.Front   = std::min(updateBox.Front,  box.Front);
+          updateBox.Back    = std::max(updateBox.Back,   box.Back);
+        }
+      } else {
+        m_updateDirtyBoxes[layer] = { 0, 0, m_desc.Width, m_desc.Height };
```

Comment:
should be 0, 0, width, height, 0, depth?

based on:

typedef struct D3DBOX {
  UINT Left;
  UINT Top;
  UINT Right;
  UINT Bottom;
  UINT Front;
  UINT Back;
} D3DBOX, *LPD3DBOX;

Reply:
Doesn't this also need to account for mip size?

Reply:
> Doesn't this also need to account for mip size?

Tracking only happens for mip 0 (according to the Microsoft documentation)

Reply:
Oh yeah, I was mistaking layer for mip and not face.

I think we should still do the 0, depth thing though.